### ISSUE TITLE
Identity loss is noticeable

### ISSUE THREAD

**Issue by 2blackbar** [sczhou/CodeFormer#13]
Hey is there a chance thats going to be improved in the model ,i can see it changing eye colours too much rom dark ones to blue if lowres image is a bit noisy in eye area for example, faces look good and realistic but something like GPEN or GFPGAN hves better identity preservation on the same face.
Its not like all upscaled faces lose identity but a lot of them do

**Comment by sczhou** [comment]
Hi, you could try to set w=1 to produce a better-fidelity result. Note that this current version is towards heavy degradation, we found that CodeFormer preserves a better identity than GPEN and GFPGAN in case of highly corrupted inputs, as shown in Figure 3 and Table 1 (IDS). If you'd like to make it on normal ones, one possible solution is retraining Stage III on smaller degradations. The training code will be released after the paper is accepted, please stay tuned.

Duplicate of #9
